### DISCUSSION TITLE
Is it intended that `SqliteConnection` does not implement `Debug`?

### DISCUSSION THREAD

**Discussion by hasezoey** [diesel-rs/diesel#3124] [Q&A] [answered]
Basically what the title says, is it intended that `SqliteConnection` does not implement the trait `Debug`? it would be helpful to have this implementation for example when having a connection function that returns `Result<SqliteConnection, SomeError>` and in tests wanting to use `unwrap_err` / `expect_err`

**Comment by weiznich** [comment, marked as answer]
There is no meaningful way to implement `Debug` for connection types. We just don't have something meaningful to print there as all of the inner fields do not implement `Debug` itself. They are mostly pointers from various C ffi crates.

**Reply by hasezoey** [reply]
> We just don't have something meaningful to print there as all of the inner fields do not implement Debug itself.

then how about just implementing debug with no fields or fields that are just pointers that are stringified? i know this might be a little weird but would help when trying to use with the mentioned use-case

**Reply by weiznich** [reply]
There is relevant discussion in https://github.com/diesel-rs/diesel/issues/2567. Generally speaking I might accept a PR adding these impl's as long as they are meaningful and do not expose potential confidential information but otherwise this is nothing I plan to work on.
